internal/factsengine/gathering_test.go

```diff
@@ -0,0 +1,196 @@
+package factsengine
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	"github.com/trento-project/agent/internal/factsengine/gatherers"
+)
+
+type GatheringTestSuite struct {
+	suite.Suite
+}
+
+func TestGatheringTestSuite(t *testing.T) {
+	suite.Run(t, new(GatheringTestSuite))
+}
+
+type DummyGatherer1 struct {
+}
+
+func NewDummyGatherer1() *DummyGatherer1 {
+	return &DummyGatherer1{}
+}
+
+func (s *DummyGatherer1) Gather(_ []gatherers.FactRequest) ([]gatherers.Fact, error) {
+	return []gatherers.Fact{
+		{
+			Name:    "dummy1",
+			Value:   "1",
+			CheckID: "check1",
+		},
+	}, nil
+}
+
+type DummyGatherer2 struct {
+}
+
+func NewDummyGatherer2() *DummyGatherer2 {
+	return &DummyGatherer2{}
+}
+
+func (s *DummyGatherer2) Gather(_ []gatherers.FactRequest) ([]gatherers.Fact, error) {
+	return []gatherers.Fact{
+		{
+			Name:    "dummy2",
+			Value:   "2",
+			CheckID: "check1",
+		},
+	}, nil
+}
+
+type ErrorGatherer struct {
+}
+
+func NewErrorGatherer() *ErrorGatherer {
+	return &ErrorGatherer{}
+}
+
+func (s *ErrorGatherer) Gather(_ []gatherers.FactRequest) ([]gatherers.Fact, error) {
+	return []gatherers.Fact{}, fmt.Errorf("kabum!") //nolint
+}
+
+func (suite *GatheringTestSuite) TestGatheringGatherFacts() {
```

Comment:
The test name sound a bit redundant :sweat_smile: 

This is the happy path, right? What about `TestFactsGathering|TestSuccessfulFactsGathering` or similar?